src/app/modules/chat/messages/messages.component.ts

```diff
@@ -1,32 +1,46 @@
-import { Component, OnInit } from '@angular/core';
+import {AfterViewChecked, Component, ElementRef, OnInit, ViewChild} from '@angular/core';
 import { Message } from '../../../models/message.interface';
 import {select, Store} from '@ngrx/store';
 import * as fromApp from '../../../store';
 import { LoadMessagesBegin, StartSendingMessage } from '../store/chat.actions';
 import { Observable, of } from 'rxjs';
 import { selectMessages } from '../store/chat.selectors';
 
+
 @Component({
   selector: 'app-home',
   templateUrl: './messages.component.html',
   styleUrls: ['./messages.component.css']
 })
-export class MessagesComponent implements OnInit {
+export class MessagesComponent implements OnInit, AfterViewChecked {
+  @ViewChild('scroll', {static: false}) private myScrollContainer: ElementRef;
   isLoading = false;
 
   public messages$: Observable<Message[]> = of([]);
 
   constructor(private store: Store<fromApp.AppState>) {}
 
   ngOnInit(): void {
+    this.scrollToBottom();
+
     this.messages$ = this.store.pipe(
       select(selectMessages)
     );
 
     this.store.dispatch(new LoadMessagesBegin());
   }
 
+  ngAfterViewChecked(): void {
+    this.scrollToBottom();
+  }
+
   addMessage(message): void {
     this.store.dispatch(new StartSendingMessage({message}));
   }
+
+  scrollToBottom(): void {
+    try {
+      this.myScrollContainer.nativeElement.scrollTop = this.myScrollContainer.nativeElement.scrollHeight;
+    } catch (err) { }
```

Comment:
Wow, not so fast 💨
Just catching errors may lead to bugs hard to discover.
What was the reason to do that?

Reply:
This function scrolls down at each event. Should I handle the error? Or this is the wrong approach?

Reply:
@MariaRoz what was that error?

Reply:
@myownsumm  In case if can’t call the scroll. But maybe the block try, catch doesn't make sense here, so I can remove it.

Reply:
Could you reproduce this error and provide a screenshot?

Reply:
@myownsumm Yes, this one:
![Screen Shot 2019-11-22 at 2 20 44 PM](https://user-images.githubusercontent.com/23140788/69425274-50e68800-0d33-11ea-8db3-0648d8e87451.png)

Reply:
@MariaRoz this may happen when content was not init yet. Did you try to use this function call not in `OnInit`() but in something like `AfterContentInit`?

Reply:
@myownsumm Hm, everything works fine with OnInit() and there are no errors.
I make this error on purpose, to show why I use block try catch. In case if will be such error and it can be caught.

Reply:
In the way you used to apply, you're will be just ignoring this error, not solving at all. Please, remove `try catch` block at all.

Reply:
Fixed